kernel/src/core_local.rs

```diff
@@ -0,0 +1,17 @@
+use core::cell::UnsafeCell;
+
+pub struct CoreLocal<T>(UnsafeCell<T>);
```

Comment:
I don't see why we need `UnsafeCell` here, though maybe the documentation will clarify that when it is added.

Reply:
We don't, I believe! I'm confident this is an artifact from our earlier design. We may want to re-introduce it in the future, if we ever add an `unsafe fn with_mut`, where developers themselves must ensure non-reentrance.